src/Symfony/Component/DependencyInjection/Compiler/ResolveTagsInheritancePass.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\ChildDefinition;
+
+/**
+ * Applies tags inheritance to definitions.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class ResolveTagsInheritancePass extends AbstractRecursivePass
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected function processValue($value, $isRoot = false)
+    {
+        if (!$value instanceof ChildDefinition || !$value->getInheritTags()) {
+            return parent::processValue($value, $isRoot);
+        }
+        $value->setInheritTags(false);
+
+        if (!$this->container->has($parent = $value->getParent())) {
```

Comment:
actually, I made it throw, so that tag inheritance is made incompatible with dynamically created parents and save a few surprises (since tags would not be inherited for them, despite the field)